core/client/fs/src/main/java/alluxio/client/file/cache/PageStore.java

```diff
@@ -45,51 +43,66 @@
    *
    * @param options the options to instantiate the page store
    * @return a PageStore instance
-   * @throws IOException if failed to create a page store
    */
   static PageStore create(PageStoreOptions options) throws IOException {
-    LOG.info("Create PageStore option={}", options.toString());
-    switch (options.getType()) {
-      case LOCAL:
-        return new LocalPageStore(options.toOptions());
-      case ROCKS:
-        return new RocksPageStore(options.toOptions());
-      default:
-        throw new IllegalArgumentException(
-            "Incompatible PageStore " + options.getType() + " specified");
-    }
+    return create(options, null, null);
   }
 
   /**
-   * Creates a new instance of {@link PageStore} based on configuration.
+   * Creates a {@link PageStore} by restoring from previous state (when metastore and evictor are
+   * not null).
    *
-   * @param conf configuration
-   * @return the {@link PageStore}
+   * @param options the options to instantiate the page store
+   * @param metaStore meta store
+   * @param evictor evictor
+   * @return a PageStore instance
+   * @throws IOException if I/O error happens
    */
-  static PageStore create(AlluxioConfiguration conf) throws IOException {
-    PageStoreOptions options;
-    PageStoreType storeType = conf.getEnum(
-        PropertyKey.USER_CLIENT_CACHE_STORE_TYPE, PageStoreType.class);
-    switch (storeType) {
+  static PageStore create(PageStoreOptions options, @Nullable MetaStore metaStore,
+        @Nullable CacheEvictor evictor) throws IOException {
+    LOG.info("Create PageStore option={}", options.toString());
+    if (metaStore == null || evictor == null) {
+      initialize(options);
+    }
+    final PageStore pageStore;
+    switch (options.getType()) {
       case LOCAL:
-        options = new LocalPageStoreOptions()
-            .setFileBuckets(conf.getInt(PropertyKey.USER_CLIENT_CACHE_LOCAL_STORE_FILE_BUCKETS));
+        pageStore = new LocalPageStore(options.toOptions());
         break;
       case ROCKS:
-        options = new RocksPageStoreOptions();
+        pageStore = new RocksPageStore(options.toOptions());
         break;
       default:
-        throw new IllegalArgumentException(String.format("Unrecognized store type %s",
-            storeType.name()));
+        throw new IllegalArgumentException(
+            "Incompatible PageStore " + options.getType() + " specified");
+    }
+    if (metaStore != null && evictor != null) {
```

Comment:
Would it be more clear if we just move the logic before this line to `create(options)` and just call it first?

Reply:
not sure I understand the comment